test/crash/src/test/java/io/narayana/lra/arquillian/LRACoordinatorRecoveryIT.java

```diff
@@ -147,60 +148,59 @@ public static WebArchive deploy() {
+    public void lraCoordinatorShortTimeoutLRAwithParticipant(
+            @ArquillianResource @OperateOnDeployment(LRA_PARTICIPANT_DEPLOYMENT_QUALIFIER) URL baseURL)
             throws URISyntaxException {
-
         String lraId;
         URI lraListenerURI = UriBuilder.fromUri(baseURL.toURI()).path(LRAListener.LRA_LISTENER_PATH).build();
-
-        // Starts an LRA with a short time limit by invoking a resource annotated with @LRA. The time limit is
+        // Starts an LRA with a short time limit by invoking a resource annotated with
+        // @LRA. The time limit is
         // defined as LRAListener.LRA_SHORT_TIMELIMIT
-        Response response = client
-                .target(lraListenerURI)
-                .path(LRAListener.LRA_LISTENER_ACTION)
-                .request()
-                .put(null);
-
+        Response response = client.target(lraListenerURI).path(LRAListener.LRA_LISTENER_ACTION).request().put(null);
         Assert.assertEquals("LRA participant action", 200, response.getStatus());
-
+        // The LRA transaction could have been started via lraClient but it is useful to
+        // test the filters as well
+        lraId = getFirstLRAFromFS();
         // Restarts lra-coordinator to simulate a crash
         restartContainer(LRA_COORDINATOR_CONTAINER_QUALIFIER);
-
         // Waits for a period of time longer than the timeout of the LRA Transaction
         doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
-        // The LRA transaction could have been started via lraClient but it is useful to test the filters as well
-        lraId = getFirstLRAFromFS();
         assertNotNull("A new LRA should have been added to the object store before the JVM was halted.", lraId);
         lraId = String.format("%s/%s", lraClient.getCoordinatorUrl(), lraId);
-
         // Waits for a period of time longer than the timeout of the LRA Transaction
         doWait((LRAListener.LRA_SHORT_TIMELIMIT + 1L) * 1000);
-
         // Checks recovery
         LRAStatus status = getStatus(new URI(lraId));
-
         LRALogger.logger.infof("%s: Status after restart is %s%n", status == null ? "GONE" : status.name());
+        Assert.assertTrue(String.format("LRA %s should have cancelled", lraId),
+                status == null || status == LRAStatus.Cancelled);
+        // Verifies that the resource was notified that the LRA finished
+        String listenerStatus = getStatusFromListener(lraListenerURI);
+        assertEquals(String.format(
+                "The service lra-listener should have been told that the final state of the LRA %s was cancelled",
+                lraId), LRAStatus.Cancelled.name(), listenerStatus);
+    }
 
-        if (status == null || status == LRAStatus.Cancelling) {
-            int sc = recover();
+    @Test
+    public void lraCoordinatorShortTimeoutLRA(@ArquillianResource @OperateOnDeployment(LRA_PARTICIPANT_DEPLOYMENT_QUALIFIER) URL baseURL)
```

Comment:
Please can I ask if this first test could be moved before the new lraCoordinatorShortTimeoutLRAwithParticipant hoping then that the diff is easier to review?

Reply:
I have done so, but I am not sure now it is better.

Reply:
It is better in my opinion - thank you!